src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs

```diff
@@ -180,7 +180,7 @@ private List<SourceRepository> GetEffectiveSourcesCore(ISettings settings, IList
                 CachingSourceProvider = new CachingSourceProvider(packageSourceProvider);
             }
 
-            return sourceObjects.Select(entry => CachingSourceProvider.CreateRepository(entry.Value)).ToList();
+            return sourceObjects.Select(entry => CachingSourceProvider.CreateRepository(entry.Value)).AsList();
```

Comment:
This change is unnecessary. There's no functional change here. 
LINQ generates enumerates which had to be converted to List.